docs/compiling.rst

```diff
@@ -100,25 +96,22 @@ regular LTO if ``-flto=thin`` is not available.
 Configuration variables
 -----------------------
 
-By default, pybind11 will compile modules with the C++14 standard, if available
-on the target compiler, falling back to C++11 if C++14 support is not
-available.  Note, however, that this default is subject to change: future
-pybind11 releases are expected to migrate to newer C++ standards as they become
-available.  To override this, the standard flag can be given explicitly in
-`CMAKE_CXX_STANDARD <https://cmake.org/cmake/help/v3.17/variable/CMAKE_CXX_STANDARD.html>`_:
+By default, pybind11 will compile modules with the compiler default or the
+minimum standard required by PyBind11, whichever is higher.  You can set the
+standard explicitly with
+`CMAKE_CXX_STANDARD <https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD.html>`_:
 
 .. code-block:: cmake
 
     # Use just one of these:
-    set(CMAKE_CXX_STANDARD 11)
     set(CMAKE_CXX_STANDARD 14)
```

Comment:
What about doing the standard formula here instead:

```cmake

set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)

```

I would expect users would be able to replace the 14 easily, and this is what you see in most CMake (non-library) projects.

Reply:
Would be OK with me (potentially, you put `set(CMAKE_CXX_STANDARD 14)  # Change to 11, 14, 17 as wanted`

Do we need the last line, specifically for pybind11? If not, you might put a comment next to it that it's optional, and what it does?

Reply:
A: Added

B: No, this is just a strictness setting - CMake will do something like `-std=g++14` without it; projects still build correctly with the compiler specific extensions. I don't think it needs an explicit comment, since you often see these together like this - but I would have added one, I just didn't remember this till after I pushed.

Reply:
And the STANDARD_REQUIRED one is optional too, it just is allowed to "decay" if your compiler doesn't support it, when  many users prefer a config error.

Reply:
Added notes.

Reply:
Thanks! :-)